kernel/kernel-api/src/test/scala/io/delta/kernel/coordinatedcommits/InMemoryCommitCoordinatorClient.scala

```diff
@@ -0,0 +1,319 @@
+        commitVersion - 1,
+        tablePath)
+      backfillToVersion(engine, tableDescriptor, commitVersion - 1, null)
+    }
+    // Write new commit file in _commits directory
+    val fileStatus = CommitCoordinatorUtils.writeUnbackfilledCommitFile(
+      engine, logPath, commitVersion, actions)
+    // Do the actual commit
+    val commitTimestamp = updatedActions.getCommitInfo.getCommitTimestamp
+    val commitResponse = addToMap(logPath, commitVersion, fileStatus, commitTimestamp)
+
+    val mcToFsConversion = CommitCoordinatorUtils.isCoordinatedCommitsToFSConversion(
+      commitVersion, updatedActions)
+    // Backfill if needed
+    if (batchSize <= 1) {
+      // Always backfill when batch size is configured as 1
+      val targetFile = backfill(engine, logPath, commitVersion, fileStatus)
+      val targetFileStatus = engine.getFileSystemClient.getFileStatus(targetFile)
+      val oldCommit = commitResponse.getCommit
+      val newCommit = new Commit(oldCommit.getVersion, targetFileStatus, oldCommit.getTimestamp)
+      return new CommitResponse(newCommit)
+    }
+    else if (commitVersion % batchSize == 0 || mcToFsConversion) {
+      logger.info(
+        "Making sure commits are backfilled till {} version for table {}",
+        commitVersion,
+        tablePath)
+      backfillToVersion(engine, tableDescriptor, commitVersion, null)
+    }
+    logger.info("Commit {} done successfully on table {}", commitVersion, tablePath)
+    commitResponse
+  }
+
+  override def getCommits(
+      engine: Engine,
+      tableDescriptor: TableDescriptor,
+      startVersion: Optional[java.lang.Long],
+      endVersion: Optional[java.lang.Long]): GetCommitsResponse = {
+    requireNonNull(engine, "engine is null")
+    requireNonNull(tableDescriptor, "tableDescriptor is null")
+    requireNonNull(startVersion, "startVersion is null")
+    requireNonNull(endVersion, "endVersion is null")
+
+    withReadLock[GetCommitsResponse](tableDescriptor.getLogPath) {
+      val tableData = perTableMap.get(tableDescriptor.getLogPath)
+      val effectiveStartVersion = startVersion.orElse(0L)
+      // Calculate the end version for the range, or use the last key if endVersion is not
+      // provided
+      val effectiveEndVersion = endVersion.orElseGet(
+        () => if (tableData.commitsMap.isEmpty) effectiveStartVersion
+        else tableData.commitsMap.lastKey)
+      val commitsInRange =
+        tableData.commitsMap.subMap(effectiveStartVersion, effectiveEndVersion + 1)
+      new GetCommitsResponse(
+        new ArrayList[Commit](commitsInRange.values), tableData.lastRatifiedCommitVersion)
+    }
+  }
+
+  override def backfillToVersion(
+      engine: Engine,
+      tableDescriptor: TableDescriptor,
+      version: Long,
+      lastKnownBackfilledVersion: Optional[java.lang.Long]): Unit = {
+    requireNonNull(engine, "engine is null")
+    requireNonNull(tableDescriptor, "tableDescriptor is null")
```

Comment:
`nonNull` check for version as well?

Reply:
Great question. In Scala, the `Long` type is a literal/primitive type so, like in Java, it cannot be null.